src/Renci.SshNet/Abstractions/DiagnosticAbstraction.cs

```diff
@@ -1,34 +1,47 @@
 ﻿using System.Diagnostics;
-#if FEATURE_DIAGNOSTICS_TRACESOURCE
-using System.Threading;
-#endif // FEATURE_DIAGNOSTICS_TRACESOURCE
 
 namespace Renci.SshNet.Abstractions
 {
-    internal static class DiagnosticAbstraction
+    /// <summary>
+    /// Diagnostics for Renci library
+    /// </summary>
+    public static class DiagnosticAbstraction
     {
 #if FEATURE_DIAGNOSTICS_TRACESOURCE
 
         private static readonly SourceSwitch SourceSwitch = new SourceSwitch("SshNetSwitch");
 
+        /// <summary>
+        /// Whether the specified event type is enabled for tracing or not
+        /// </summary>
+        /// <param name="traceEventType">The trace event type</param>
+        /// <returns>true if enabled for tracing, false otherwise</returns>
         public static bool IsEnabled(TraceEventType traceEventType)
         {
             return SourceSwitch.ShouldTrace(traceEventType);
         }
 
-        private static readonly TraceSource Loggging =
+        /// <summary>
+        /// The trace source for Renci
+        /// </summary>
+        public static readonly TraceSource Logging =
 #if DEBUG
-            new TraceSource("SshNet.Logging", SourceLevels.All);
+            new TraceSource(name: "SshNet.Logging", defaultLevel: SourceLevels.All);
 #else
             new TraceSource("SshNet.Logging");
 #endif // DEBUG
 #endif // FEATURE_DIAGNOSTICS_TRACESOURCE
 
-        [Conditional("DEBUG")]
```

Comment:
Please make it conditional for now.
We'll need to review the impact on performance.

Reply:
That was the whole point of the PR, and the bug filed against this product years ago, to enable some tracing for retail build. There isn't much substance without that. I guess this problem then need solving through modernization of the tracing approach for SSH .NET as a whole. For that, well-known and broadly adopted ETW tracing should likely be preferred.